stdlib/messaging/kafka/src/main/ballerina/src/kafka/producer.bal

```diff
@@ -0,0 +1,154 @@
+# + linger - Delay to allow other records to be batched.
+# + sendBuffer - Size of the TCP send buffer (SO_SNDBUF).
+# + receiveBuffer - Size of the TCP receive buffer (SO_RCVBUF).
+# + maxRequestSize - The maximum size of a request in bytes.
+# + reconnectBackoff - Time to wait before attempting to reconnect.
+# + reconnectBackoffMax - Maximum amount of time in milliseconds to wait when reconnecting.
+# + retryBackoff - Time to wait before attempting to retry a failed request.
+# + maxBlock - Maximum block time which the send is blocked, when the buffer is full.
+# + requestTimeout - Wait time for response of a request.
+# + metadataMaxAge - Maximum time to force a refresh of metadata.
+# + metricsSampleWindow - Time window for a metrics sample to computed over.
+# + metricsNumSamples - Number of samples maintained to compute metrics.
+# + maxInFlightRequestsPerConnection - Maximum number of unacknowledged requests on a single connection.
+# + connectionsMaxIdle - Close idle connections after the number of milliseconds.
+# + transactionTimeout - Timeout for transaction status update from the producer.
+# + enableIdempotence - Exactly one copy of each message is written in the stream when enabled.
+# + secureSocket - SSL/TLS related options
+public type ProducerConfig record {|
+    string? bootstrapServers = (); // BOOTSTRAP_SERVERS_CONFIG 0
+    string? acks = (); // ACKS_CONFIG 1
+    string? compressionType = (); // COMPRESSION_TYPE_CONFIG 2
+    string? clientId = (); // CLIENT_ID_CONFIG 3
+    string? metricsRecordingLevel = (); // METRICS_RECORDING_LEVEL_CONFIG 4
+    string? metricReporterClasses = (); // METRIC_REPORTER_CLASSES_CONFIG 5
+    string? partitionerClass = (); // PARTITIONER_CLASS_CONFIG 6
+    string? interceptorClasses = (); // INTERCEPTOR_CLASSES_CONFIG 7
+    string? transactionalId = (); // TRANSACTIONAL_ID_CONFIG 8
+
+    int bufferMemory = -1; // BUFFER_MEMORY_CONFIG 0
+    int noRetries = -1; // RETRIES_CONFIG 1
+    int batchSize = -1; // BATCH_SIZE_CONFIG 2
+    int linger = -1; // LINGER_MS_CONFIG 3
+    int sendBuffer = -1; // SEND_BUFFER_CONFIG 4
+    int receiveBuffer = -1; // RECEIVE_BUFFER_CONFIG 5
+    int maxRequestSize = -1; // MAX_REQUEST_SIZE_CONFIG 6
+    int reconnectBackoff = -1; // RECONNECT_BACKOFF_MS_CONFIG 7
+    int reconnectBackoffMax = -1; // RECONNECT_BACKOFF_MAX_MS_CONFIG  8
+    int retryBackoff = -1; // RETRY_BACKOFF_MS_CONFIG 9
+    int maxBlock = -1; // MAX_BLOCK_MS_CONFIG 10
+    int requestTimeout = -1; // REQUEST_TIMEOUT_MS_CONFIG  11
+    int metadataMaxAge = -1; // METADATA_MAX_AGE_CONFIG 12
+    int metricsSampleWindow = -1; // METRICS_SAMPLE_WINDOW_MS_CONFIG 13
+    int metricsNumSamples = -1; // METRICS_NUM_SAMPLES_CONFIG  14
+    int maxInFlightRequestsPerConnection = -1; // MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION 15
+    int connectionsMaxIdle = -1; // CONNECTIONS_MAX_IDLE_MS_CONFIG 16
+    int transactionTimeout = -1; // TRANSACTION_TIMEOUT_CONFIG 17
```

Comment:
I guess these comments represent the java level constant names and indices. Are indices required anymore? Don't we obtain the values by name?

Reply:
Removed.